lib/Crypto/CryptHSM.php

```diff
@@ -168,7 +168,7 @@ public function multiKeyDecrypt($encKeyFile, $shareKey, $privateKey) { // done w
 
 			// now decode the file.
 			// version and position are 0 because we always use fresh random data as passphrase
-			$decryptedContent = $this->symmetricDecryptFileContent($encKeyFile, $decryptedKey, self::DEFAULT_CIPHER, 0, 0, true);
```

Comment:
Should we connect the boolean to the `encryption.use_legacy_encoding` option? https://github.com/owncloud/core/blob/d1f0c43d61a2583fdf0f457bb07ea49de100d89a/config/config.sample.php#L1405

@butonic @karakayasemi

Reply:
If binary encoding is breaking HSM and if we will always use it with base64 encoding, no need to add that option.

But I guess we can try to understand why HSM is failing with binary encoding. Is it possible to try HSM in the development environment? If there is any documentation, I can try it.

Reply:
Not sure about the documentation, but from what I understood the issue might happen because the old 10.5 instance was using the legacy format (we also changed that back as it defaults to false), and additionally this value had to be patched in the code to make things work.